DotNetHowToSecurityTrimming/DotNetHowToSecurityTrimming/Program.cs

```diff
@@ -64,7 +67,7 @@ static void Main(string[] args)
 
             Console.WriteLine("Indexing documents...\n");
             // Index documents with relevant group ids
-            IndexDocuments(indexName, groups);
+            IndexDocuments(indexName, groups.Keys.Select(g => g.Id).ToList());
 
             foreach (var user in UsersPrincipalNameGroup)
```

Comment:
Can you use the users list extracted from the groups collection instead, so we can remove the UsersPrincipalNameGroup field?